plasmapy/classes/species.py

```diff
@@ -83,23 +83,29 @@ def __init__(self, plasma, particle_type='p', n_particles=1, scaling=1,
                                          dtype=float) * u.m
         self.velocity_history = np.zeros((self.NT, *self.v.shape),
                                          dtype=float) * (u.m / u.s)
+        # create intermediate array of dimension (nx,ny,nz,3) in order to allow
+        # interpolation on non-equal spatial domain dimensions
+        _B = np.moveaxis(self.plasma.magnetic_field.si.value, 0, -1)
+        _E = np.moveaxis(self.plasma.electric_field.si.value, 0, -1)
+
         self._B_interpolator = RegularGridInterpolator(
             (self.plasma.x.si.value,
              self.plasma.y.si.value,
              self.plasma.z.si.value),
-            self.plasma.magnetic_field.T.si.value,
+            _B,
             method="linear",
             bounds_error=True)
 
         self._E_interpolator = RegularGridInterpolator(
             (self.plasma.x.si.value,
              self.plasma.y.si.value,
              self.plasma.z.si.value),
-            self.plasma.electric_field.T.si.value,
+            _E,
             method="linear",
             bounds_error=True)
 
     def _interpolate_fields(self):
+        print(self.x.si.value)
```

Comment:
Won't this print for every iteration? Possibly not what the user wants, particularly for large grids.

Reply:
Oh yeah, that it will. Missed this one.

Reply:
Yes, it was for debug purpose. I'll remove it.